lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -339,80 +269,43 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
     } finally {
       perf.latencyInMs = Date.now() - perf.startTimestampInMs
       this.checkLastCallPerformance(perf)
-      this.updateHealthyStatus()
       if (this.enableDbSync) {
-        if (
-          !this.syncingDb &&
-          MAJOR_METHOD_NAMES.includes(perf.methodName) &&
-          this.hasEnoughWaitSinceLastDbSync(1000 * this.config.DB_SYNC_INTERVAL_IN_S)
-        ) {
+        if (!this.syncingDb && this.hasEnoughWaitSinceLastDbSync(1000 * this.config.DB_SYNC_INTERVAL_IN_S)) {
           this.logDbSyncRequested()
           if (Math.random() < this.dbSyncSampleProb) {
             this.logDbSyncSampled()
             this.syncingDb = true
             // Fire and forget. Won't check the sync result.
-            this.syncAndUpdateProviderState()
+            this.syncAndUpdateProviderHealthState()
           }
         }
       }
-      this.lastCallTimestampInMs = perf.startTimestampInMs
     }
   }
 
-  private async syncAndUpdateProviderState() {
+  private async syncAndUpdateProviderHealthState() {
     try {
-      const newState: ProviderState | null = await this.providerStateSyncer.syncWithRepository(
-        this.healthScore - this.healthScoreAtLastSync,
-        this.healthScore,
-        this.lastEvaluatedLatencyInMs,
-        this.lastLatencyEvaluationTimestampInMs,
-        this.lastLatencyEvaluationApiName
-      )
-      if (newState !== null) {
-        // Update health state
-        this.healthScoreAtLastSync = newState.healthScore
-        this.healthScore = this.healthScoreAtLastSync
-        this.log.debug(`${this.providerId}: Synced with storage: new health score ${this.healthScore}`)
-        this.updateHealthyStatus()
-
-        // Update latency stat
-        this.updateLatencyStat(newState)
+      const healthStateFromDb = await this.healthStateRepository.read(this.providerId)
```

Comment:
note for future improvement: each chain could have several alarm (P50 latency, P90 latency, success rate, volume...) mapped to columns in the DB. From this we could compute a score that gives you the weight directly here

Reply:
Thanks for the suggestion. Currently the DB schema looks like this:

```json
{
  "chainIdProviderName": {
    "S": "11155111_ALCHEMY"
  },
  "healthiness": {
    "S": "UNHEALTHY"
  },
  "ongoingAlarms": {
    "L": [
      {
        "S": "RoutingAPI-RpcGateway-LatencyAlarm-ChainId-11155111-Provider-ALCHEMY"
      }
    ]
  },
  "version": {
    "N": "29"
  }
}
```

So all ongoing alarms have been put inside the DB entry.